src/Repositories/GoalRepository.cs

```diff
@@ -1,99 +1,110 @@
-
-        using var connection = await _dbConnection.CreateConnectionAsync();
-        var affectedRows = await connection.ExecuteAsync(query, item);
-
-        return affectedRows > 0;
-    }
-}
+
+using Domain.Entities;
+
+using ExpenseTrackerGroup3.Repositories.Interfaces;
+using ExpenseTrackerGroup3.Data.Interfaces;
+using Dapper;
+
+namespace ExpenseTrackerGroup3.Repositories;
+
+public class GoalRepository : IGoalRepository
+{
+    private readonly IDbConnectionFactory _dbConnection;
+
+    public GoalRepository(IDbConnectionFactory dbConnectionFactory)
+    {
+        _dbConnection = dbConnectionFactory;
+    }
+
+    public async Task<bool> CreateAsync(Goal item)
+    {
+        const string query = @"
+        INSERT INTO Goal (Id, UserId, GoalAmount, Deadline, CurrentAmount, CreatedAt)
+        VALUES (@Id, @UserId, @GoalAmount, @DeadLine, @CurrentAmount, @CreatedAt)
+        ";
+
+        using var connection = await _dbConnection.CreateConnectionAsync();
+        var affectedRows = await connection.ExecuteAsync(query, item);
+
+        return affectedRows > 0;
+    }
+
+
+    public async Task<bool> DeleteAsync(Guid id)
+    {
+        const string query = "DELETE FROM Goal WHERE Id = @Id";
+
+        using var connection = await _dbConnection.CreateConnectionAsync();
+        var affectedRows = await connection.ExecuteAsync(query, new { Id = id });
+
+        return affectedRows > 0;
+    }
+
+    public async Task<IEnumerable<Goal>> GetActiveGoalsByUserIdAsync(Guid userId)
+    {
+        const string query = @"
+        SELECT * FROM Goal 
+        WHERE UserId = @UserId 
+        AND DeadLine > NOW()
+        ";
+
+        using var connection = await _dbConnection.CreateConnectionAsync();
+        return await connection.QueryAsync<Goal>(query, new { UserId = userId });
+    }
+
+    public async Task<IEnumerable<Goal>> GetAllAsync()
+    {
+        const string query = "SELECT * FROM Goal";
+
+        using var connection = await _dbConnection.CreateConnectionAsync();
+
+        return await connection.QueryAsync<Goal>(query);
+    }
+
+    public async Task<Goal?> GetByIdAsync(Guid id)
+    {
+        const string query = "SELECT * FROM Goal WHERE Id = @Id";
+
+        using var connection = await _dbConnection.CreateConnectionAsync();
+        return await connection.QuerySingleOrDefaultAsync<Goal>(query, new { Id = id });
+    }
+
+    public async Task<IEnumerable<Goal>> GetGoalsByUserIdAsync(Guid userId)
+    {
+        const string query = "SELECT * FROM Goal WHERE UserId = @UserId";
+
+        using var connection = await _dbConnection.CreateConnectionAsync();
+        var goals = await connection.QueryAsync<Goal>(query, new { UserId = userId });
+        return goals.ToList(); 
+    }
+
+    public async Task<decimal> GetGoalProgressAsync(Guid id)
+    {
+        const string query = @"
+        SELECT 
+            CASE 
+                WHEN GoalAmount = 0 THEN 0 
+                ELSE (CurrentAmount / GoalAmount) 

```

Comment:
Please fix the operation to calculate the goal progress, is given wrong results, in this case the goal was 20 and the current amount 1, which should give 5% but it gives me 0.05:
![Screenshot from 2024-09-26 07-36-52](https://github.com/user-attachments/assets/1f06e4b1-6c60-4edb-89ba-d0ef8030d103)